### PR TITLE
Adding baseurl flag to allow self-hosted GitLab/GitHub servers

### PR DESCRIPTION
**Pull request by petegallagher**
I have tested this with a self-hosted GitLab instance. I have not tested this with a GitHub Enterprise installation, but it *should* work...

### CODE REVIEW

File: hosts/github.go [hunk 1]
```diff
@@ -33,9 +33,22 @@ func NewGithubClient(m manager.Manager) *Github {
...
+		githubClient = github.NewClient(httpClient)
+	} else {
+		var err error
+		githubClient, err = github.NewEnterpriseClient(m.Opts.BaseURL, m.Opts.BaseURL, httpClient)
+		if err != nil {
+			log.Error(err)
+		}
+	}
+
 	return &Github{
```
Comment (zricethezav) [inline hosts/github.go]: This all looks good but let's add a nil check here https://github.com/zricethezav/gitleaks/blob/master/hosts/host.go#L31. Something like 
```
	if host == nil {
		return fmt.Errorf("unable to create %s client", getHost(m.Opts.Host))
	}
```

Comment (zricethezav) [reply hosts/github.go]: Or we could change the function signature to return an error

Comment (petegallagher) [reply hosts/github.go]: Yeah, makes a lot more sense to return the errors.

File: hosts/github.go [hunk 2]
```diff
@@ -27,16 +27,29 @@ type Github struct {
...
 
+	var githubClient *github.Client
+	httpClient := oauth2.NewClient(ctx, token)
+
+	if m.Opts.BaseURL == "" {
+		githubClient = github.NewClient(httpClient)
+	} else {
+		githubClient, err = github.NewEnterpriseClient(m.Opts.BaseURL, m.Opts.BaseURL, httpClient)
+		if err != nil {
+			log.Error(err)
```
Comment (zricethezav) [inline hosts/github.go]: This check could be removed since the error will be bubbled up in main.go

File: hosts/gitlab.go [hunk 3]
```diff
@@ -22,12 +22,23 @@ type Gitlab struct {
...
+	gitlabClient := &Gitlab{
 		manager: m,
 		ctx:     context.Background(),
 		client:  gitlab.NewClient(nil, options.GetAccessToken(m.Opts)),
 	}
+
+	if m.Opts.BaseURL != "" {
+		err := gitlabClient.client.SetBaseURL(m.Opts.BaseURL)
+		if err != nil {
+			log.Error(err)
```
Comment (zricethezav) [inline hosts/gitlab.go]: This check could be removed since the error will be bubbled up in main.go

File: hosts/github.go (hunk 2 above)
Comment (petegallagher) [reply hosts/github.go]: Updated

File: hosts/gitlab.go (hunk 3 above)
Comment (petegallagher) [reply hosts/gitlab.go]: Updated

**Pull request merged**